[stars-img]: https://img.shields.io/github/stars/DayuHuu/Awesome-Single-Cell-Clustering?color=yellow
[stars-url]: https://github.com/DayuHuu/Awesome-Single-Cell-Clustering/stargazers
[fork-img]: https://img.shields.io/github/forks/DayuHuu/Awesome-Single-Cell-Clustering?color=lightblue&label=fork
[fork-url]: https://github.com/DayuHuu/Awesome-Single-Cell-Clustering/network/members
[AKGR-url]: https://github.com/DayuHuu/Awesome-Single-Cell-Clustering

# Awesome-Single-Cell-Clustering
ASCC: Awesome Single Cell Clustering is a collection of single-cell clustering works, including papers, codes and datasets :fire:. Any problems, please contact hzauhdy@gmail.com or w13840532920@163.com. Any other interesting papers or codes are welcome. If you find this repository useful to your research or work, it is really appreciated to star this repository.

[![GitHub stars][stars-img]][stars-url]
[![GitHub forks][fork-img]][fork-url]

<div  align="center">    
    <img src="./flow.png" width=100% />
</div>


### Embedded Single-Cell Deep Clustering

| Year | Title                                                        |  Venue  |                            Paper                             |                             Code                             |
| ---- | ------------------------------------------------------------ | :-----: | :----------------------------------------------------------: | :----------------------------------------------------------: |
| 2024 | **（scDFN）enhancing single-cell RNA-seq clustering with deep fusion networks** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbae486/7811489)| - |
| 2024 | **Dimension reduction, cell clustering, and cell–cell communication inference for single-cell transcriptomics with DcjComm** | GB | [link](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-024-03385-6)| - |
| 2024 | **(G3DC)A Gene-Graph-Guided Selective Deep Clustering Method for Single Cell RNA-seq Data** | BDMAA | [link](https://ieeexplore.ieee.org/document/10654682/)| - |
| 2024 | **(scASDC)Attention Enhanced Structural Deep Clustering for Single-cell RNA-seq Data** | IEEE | [link](http://arxiv.org/abs/2408.05258)| - |
| 2024 | **(DP-DCAN)Differentially Private Deep Contrastive Autoencoder Network for Single-cell Clustering** | - | [link](http://arxiv.org/abs/2311.03410)| - |
| 2024 | **(scMNMF)a novel method for single-cell multi-omics clustering based on matrix factorization** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbae228/7675151)| - |
| 2023 | **A critical assessment of clustering algorithms to improve cell clustering and identification in single-cell transcriptome study** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbad497/7505072)| - |
| 2023 | **Attention-based deep clustering method for scRNA-seq cell type identification** | CB | [link](https://dx.plos.org/10.1371/journal.pcbi.1011641)| - |
| 2023 | **Self-supervised deep clustering of single-cell RNA-seq data to hierarchically detect rare cell populations** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbad335/7283019)| - |
| 2023 | **（DISCERN）deep single-cell expression reconstruction for improved cell clustering and cell subtype and state detection**| GB | [link](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-03049-x)| - |
| 2023 | **Deep enhanced constraint clustering based on contrastive learning for scRNA-seq data** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbad222/7197084)| - |
| 2023 | **（scDCCA）deep contrastive clustering for single-cell RNA-seq data based on auto-encoder network** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbac625/6984787)| - |
| 2022 | **[ADClust] A parameter-free deep embedded clustering method for single-cell RNA-seq data**  | BIB | [link](https://academic.oup.com/bib/article/23/5/bbac172/6582003?login=false) | [link](https://github.com/biomed-AI/ADClust) | 
| 2022 | **(scDSSC)Deep Sparse Subspace Clustering for scRNA-seq Data** | CB | [link](https://dx.plos.org/10.1371/journal.pcbi.1010772)| - |
| 2022 | **(NISC)Neural Network-Imputation for Single-Cell RNA Sequencing and Cell Type Clustering** | OR | [link](https://www.frontiersin.org/articles/10.3389/fgene.2022.847112/full)| - |
| 2022 | **Deep clustering of small molecules at large-scale via variational autoencoder embedding and K-means**| BMC | [link](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-022-04667-1)| - |
| 2022 | **comparison of deep learning-based pre-processing  and clustering approaches for single-cell RNA  sequencing data** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbab345/6361043) | - |
| 2022 | **（ScCAEs）deep clustering of single-cell RNA-seq via convolutional autoencoder embedding and soft K-means** | BIB | [link](https://academic.oup.com/bib/article/doi/10.1093/bib/bbab321/6362111)| - |
| 2021 | **Contrastive self-supervised clustering of scRNA-seq data** | BMC | [link](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04210-8)| - |
| 2021 | **Clustering single cell CITE-seq data with a canonical correlation based deep learning method** | - | [link](http://biorxiv.org/lookup/doi/10.1101/2021.09.07.459236)| - |
| 2021 | **Deep learning-based clustering approaches for bioinformatics** | BIB | [link](https://academic.oup.com/bib/article/22/1/393/5721075)| - |
| 2020 | **（scAIDE）clustering of large-scale single-cell RNA-seq data reveals putative and rare cell types** | NAR | [link](https://academic.oup.com/nargab/article/doi/10.1093/nargab/lqaa082/5920393)| - |
| 2020 | **(scVAE)variational auto-encoders for single-cell gene expression data** | BIB | [link](https://academic.oup.com/bioinformatics/article/36/16/4415/5838187)| - |
| 2020 | **Robust unsupervised dimensionality reduction based on feature clustering for single-cell imaging data** | ASCJ | [link](https://linkinghub.elsevier.com/retrieve/pii/S1568494620303616) | - |
| 2020 | **Deep learning enables accurate clustering with batch effect removal in single-cell RNA-seq analysis** | N | [link](https://www.nature.com/articles/s41467-020-15851-3)| - |
| 2020 | **Single-Cell Transcriptome Data Clustering via Multinomial Modeling and Adaptive Fuzzy K-Means Algorithm** | OR | [link](https://www.frontiersin.org/article/10.3389/fgene.2020.00295/full)| - |
| 2020 | **Dimension Reduction and Clustering Models for Single-Cell RNA Sequencing Data: A Comparative Study** | MS | [link](https://www.mdpi.com/1422-0067/21/6/2181) | - |
| 2020 | **Pathway-based deep clustering for molecular subtyping of cancer**| HHS | [link](https://linkinghub.elsevier.com/retrieve/pii/S1046202319300489)| - |
| 2020 | **（DSCD）A Novel Deep Subspace Clustering Denoise Network for Single-Cell Clustering** | IEEE | [link](https://ieeexplore.ieee.org/document/9115662/)| - |
| 2019 | **Impact of similarity metrics on single-cell RNA-seq data clustering** | BIB | [link](https://academic.oup.com/bib/article/20/6/2316/5077112)| - |
| 2019 | **Combining gene ontology with deep neural networks to enhance the clustering of single cell RNA-Seq data** | BMC | [link](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-2769-6)| - |
| 2018 | **Clustering single cells: a review of approaches on high-and low-depth single-cell RNA-seq data**| BIFG | [link](https://academic.oup.com/bfg/article/17/4/240/4728639) | - |
| 2016 | **Single-Channel Multi-Speaker Separation using Deep Clustering** | - | [link](http://arxiv.org/abs/1607.02173)| - |


### Topological Single-Cell Deep Clustering

| Year | Title                                                        |  Venue  |                            Paper                             |                             Code                             |
| ---- | ------------------------------------------------------------ | :-----: | :----------------------------------------------------------: | :----------------------------------------------------------: |
| 2022 | **[scGAC] scGAC: a graph attentional architecture for clustering single-cell RNA-seq data**  | BIO | [link](https://academic.oup.com/bioinformatics/article/38/8/2187/6530275?login=false) | [link](https://github.com/Joye9285/scGAC) | 
| 2023 | **Toward Convex Manifolds: A Geometric Perspective for Deep Graph Clustering of Single-cell RNA-seq Data** | IJVAI | [link](https://www.ijcai.org/proceedings/2023/540) |
| 2023 | **[scGPCL] scGPCL: Deep single-cell RNA-seq data clustering with graph prototypical contrastive learning** | BIB | [link](https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btad342/7180270) |
| 2023 | **[CPSmerge] CPSmerge: Multi-view clustering by CPS-merge analysis with application to multimodal single-cell data** | CB | [link](https://dx.plos.org/10.1371/journal.pcbi.1011044) |
| 2022 | **[DGCyTOF] DGCyTOF: Deep learning with graphic cluster visualization to predict cell types of single cell mass cytometry data** | CB | [link](https://dx.plos.org/10.1371/journal.pcbi.1008885) |
| 2020 | **[FastPG] FastPG: Fast clustering of millions of single cells** | - |
| 2019 | **[PAGA] PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells** | GB | [link](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1663-x) |


### Multi-Modal Single-Cell Deep Clustering

| Year | Title                                                        |  Venue  |                            Paper                             |                             Code                             |
| ---- | ------------------------------------------------------------ | :-----: | :----------------------------------------------------------: | :----------------------------------------------------------: |
| 2022 | **[scMDC] Clustering of single-cell multi-omics data with a multimodal deep learning method**  | NC | [link](https://www.nature.com/articles/s41467-022-35031-9) | [link](https://github.com/xianglin226/scMDC) | 

